### PR TITLE
add velero deployment to CSV so velero SA/roles are added to CSV then removing it

### PR DESCRIPTION
**Pull request by kaovilai**
This fixes `make bundle` not adding velero roles/SA to CSV.

Velero placeholder deployment is added to config so `operator-sdk generate bundle` add cluster permission rules for velero SA in CSV.
`yq` is used to remove placeholder deployment from final bundle CSV.

### CODE REVIEW

File: config/velero/velero.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# This is a placeholder deployment config so operator-sdk add cluster-roles for velero to ClusterServiceVersion
+# It will be removed by yq during make bundle
+kind: Deployment
```
Comment (djzager) [inline config/velero/velero.yaml:3]: My only recommendation for this file is to strip it to make it less something that needs to be maintained.

As I understand it, the only thing that matters is that the deployment's serviceAccountName.

**Review by djzager — APPROVED** [review]
I personally prefer this to manually maintaining the csv. Have an ack.

File: config/velero/velero.yaml (hunk 1 above)
Comment (kaovilai) [reply config/velero/velero.yaml:3]: @djzager just pulled this from a running development deployment.. nothing much to maintain. Just need a container to be a valid deployment

**Comment by dymurray** [comment]
This is definitely a weird way to solve things, but it gets the job done for now. We are going to want to revisit this at some point

**Pull request merged**